x-pack/plugins/ml/public/application/jobs/new_job/common/job_creator/util/filter_runtime_mappings.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { RuntimeMappings } from '../../../../../../../common/types/fields';
+import type { Datafeed, Job } from '../../../../../../../common/types/anomaly_detection_jobs';
+
+interface Response {
+  runtime_mappings: RuntimeMappings;
+  discarded_mappings: RuntimeMappings;
+}
+
+export function filterRuntimeMappings(job: Job, datafeed: Datafeed): Response {
+  if (datafeed.runtime_mappings === undefined) {
+    return {
+      runtime_mappings: {},
+      discarded_mappings: {},
+    };
+  }
+
+  const usedFields = findFieldsInJob(job, datafeed);
+
+  const { runtimeMappings, discardedMappings } = createMappings(
+    datafeed.runtime_mappings,
+    usedFields
+  );
+
+  return { runtime_mappings: runtimeMappings, discarded_mappings: discardedMappings };
+}
+
+function findFieldsInJob(job: Job, datafeed: Datafeed) {
+  const usedFields = new Set<string>();
+  job.analysis_config.detectors.forEach((d) => {
+    if (d.field_name !== undefined) {
+      usedFields.add(d.field_name);
+    }
+    if (d.by_field_name !== undefined) {
+      usedFields.add(d.by_field_name);
+    }
+    if (d.over_field_name !== undefined) {
+      usedFields.add(d.over_field_name);
+    }
+    if (d.partition_field_name !== undefined) {
+      usedFields.add(d.partition_field_name);
+    }
+  });
```

Comment:
Consider using `getEntityFieldName` from `anomaly_utils`.

```suggestion
const usedFields = new Set<string>(ob.analysis_config.detectors.map(getEntityFieldName))
```

Reply:
`getEntityFieldName` does not return the `field_name`, only the by, over an partition fields